src/fsharp/LexFilter.fs

```diff
@@ -868,7 +861,7 @@ type LexFilterImpl (lightSyntaxStatus:LightSyntaxStatus, compilingFsLib, lexer,
                     let lookaheadTokenStartPos = startPosOfTokenTup lookaheadTokenTup
                     match lookaheadToken with 
                     | Parser.EOF _ | SEMICOLON_SEMICOLON -> false 
-                    | _ when indentation && lookaheadTokenStartPos < tokenEndPos -> false
+                    | _ when indentation && lookaheadTokenStartPos.AbsoluteOffset < tokenEndPos.AbsoluteOffset -> false
```

Comment:
Wow, the code before looks very wrong. It was effectively doing a comparison on the type that contained file indices and others, which was probably not intended.